apps/vc-api/src/did/did.controller.ts

```diff
@@ -15,39 +15,53 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { Body, Controller, Get, Param, Post } from '@nestjs/common';
+import { BadRequestException, Body, Controller, Get, NotFoundException, Param, Post } from '@nestjs/common';
 import { DIDService } from './did.service';
-import { DIDDocument, VerificationMethod } from 'did-resolver';
-import { ApiTags } from '@nestjs/swagger';
+import {
+  ApiBody,
+  ApiCreatedResponse,
+  ApiNotFoundResponse,
+  ApiOkResponse,
+  ApiOperation,
+  ApiTags
+} from '@nestjs/swagger';
 import { CreateDidOptionsDto } from './dto/create-did-options.dto';
 import { DidMethod } from './types/did-method';
+import { CreateDidResponseDto } from './dto/create-did-response.dto';
 
 @ApiTags('did')
 @Controller('did')
 export class DIDController {
   constructor(private didService: DIDService) {}
 
-  /**
-   * Generate a new DID
-   * @param body options for DID creation
-   * @returns DIDDocument of new DID
-   */
   @Post()
-  async create(@Body() body: CreateDidOptionsDto): Promise<DIDDocument> {
+  @ApiBody({ type: CreateDidOptionsDto })
+  @ApiCreatedResponse({ type: CreateDidResponseDto })
+  @ApiOperation({ description: 'Generate a new DID' })
+  async create(@Body() body: CreateDidOptionsDto): Promise<CreateDidResponseDto> {
```

Comment:
I think it is better to fix missing error responses in another PR. This is a huge change, so it will be easier to isolate the reason if something goes wrong.